scripts/license/add_license_header_test.go

```diff
@@ -0,0 +1,201 @@
+package main
+
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func Test_Recognizes_GethLicense(t *testing.T) {
+	// make a temporary file
+	tmpDir := t.TempDir()
+	originalFile, err := os.Create(tmpDir + "/test_geth_license")
+	require.NoError(t, err)
+	defer func() { _ = originalFile.Close() }()
```

Comment:
```suggestion
	defer originalFile.Close()
```

Reply:
one of our linters is `errcheck` which checks that all functions that return an error assign it to something. So we need to assignt it event if we are going to ignore it. That said, most of these defers will go away when I apply your suggestion of using `[os.WriteFile](https://pkg.go.dev/os#WriteFile)`.